xgi/classes/hypergraph.py

```diff
@@ -1152,6 +1154,35 @@ def remove_isolates(self, ignore_singletons=True):
         self.remove_nodes_from(self.isolates(ignore_singletons))
         return self
 
+    def remove_duplicates(self): 
+        """Remove all duplicate edges so that each edge
+        is unique. 
+        
+        See also
+        --------
+        remove_isolates
+        remove_singleton_edges
+        """
+        
+        edges = [tuple(e) for e in self._edge.values()]
+        edges_unique = set(edges) 
+        
+        self._edge.clear() # remove all edges
+        self._edge_attr.clear()
+        self.add_edges_from(edges_unique) # re-add only unique ones
```

Comment:
Suggestion:
```suggestion
        edges_unique = {tuple(e) for e in self._edge.values()}
        self._edge.clear()
        self._edge_attr.clear()
        self.add_edges_from(edges_unique)
```

Reply:
Also, I believe this is going to destroy all attributes and never add them back again! You'd have to filter the attributes we need to keep:
```python
...
attrs_unique = {id: self._edge_attr[id] for id in edge_unique}
..
self.add_edges_from(edges_unique, attrs_unique)
```

Reply:
Having said that, we really need a test for this that keeps track of the attributes correctly...

Reply:
Very good point, I realised that when writing it, and then never solved it. I wrote it originally to use it in the SimplicialComplex class, but in the end we use frozensets so we don't need it anymore. Deleting for now.